src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -560,16 +560,17 @@ ModifyPartialQuerySupported(Query *queryTree, bool multiShardQuery,
 			CommonTableExpr *cte = (CommonTableExpr *) lfirst(cteCell);
 			Query *cteQuery = (Query *) cte->ctequery;
 
-			if (cteQuery->commandType != CMD_SELECT)
+			/* CTEs still not supported for INSERTs. */
+			if (queryTree->commandType == CMD_INSERT)
 			{
-				/* Modifying CTEs still not supported for INSERTs & multi shard queries. */
-				if (queryTree->commandType == CMD_INSERT)
-				{
-					return DeferredError(ERRCODE_FEATURE_NOT_SUPPORTED,
-										 "Router planner doesn't support non-select common table expressions with non-select queries.",
-										 NULL, NULL);
-				}
+				return DeferredError(ERRCODE_FEATURE_NOT_SUPPORTED,
+									 "Router planner doesn't support common table expressions with INSERT queries.",
+									 NULL, NULL);
+			}
 
+			if (cteQuery->commandType != CMD_SELECT)
+			{
+				/* Modifying CTEs still not supported for multi shard queries. */
```

Comment:
just to understand better, is this part of the bugfix or some style & error/debug message improvement ?

Reply:
As mentioned in the description, It is part of the bugfix for this case:
```SQL
with wcte as (select * from table1)
  insert into table2 values ('hello world');
```

Reply:
now I understood the difference, I was confused since I couldn't realize that we check for different things in those two if's (`cteQuery->commandType` and `queryTree->commandType`)